tests/saw/spec/handshake/s2n_handshake_io.cry

```diff
@@ -21,15 +21,19 @@ conn_set_handshake_type conn = conn'
                 }
         (handshake' : handshake) = {handshake_type = handshake_type'
                                    ,message_number = conn.handshake.message_number
+                                   ,hello_retry_request = conn.handshake.hello_retry_request
                                    }
-        handshake_type' = if (IS_TLS13_HANDSHAKE conn) then NEGOTIATED || FULL_HANDSHAKE
+        handshake_type' = if (IS_TLS13_HANDSHAKE conn) then NEGOTIATED || FULL_HANDSHAKE || hello_retry_request
```

Comment:
below this is HELLO_RETRY_REQUEST (L35 and L251). I believe it should be here as well

Reply:
This is similar to `full_handshake`. Its value is resolved based on other conditions in the handshake.